lib/simplekiq/orchestration_job.rb

```diff
@@ -21,6 +28,16 @@ def workflow_plan(*args)
 
     private
 
+    def conditionally_within_parent_batch
```

Comment:
Not a big deal but is there reasoning behind having this live in the job class rather than the executor class?

Reply:
@jacknoble  A few things:
* this "conditionality" was previously communicated to the executor via overloading the `parent_batch` variable with sometimes having `nil` which is difficult to be defensive about whether the variable was actually intended to be nil as an implicit message or whether nil slipped in unintentionally
* rather than passing the parent batch from this class to the executor and requiring them both to have a dependency on parent batch nesting, instead we can handle it here and have the executor not need to know a thing about parent batches and just leverage sidekiq-pro batching behavior to maintain that connection between the toplevel orchestration batch and then parent batch, and just have the executor focus on maintaining the connection between the step batches and the orchestration batch.
* This also came with moving the dependency on `OrchestrationExecutor.execute` from `Orchestration` (the workflow structure) instead into this class, the job mixin. Together, this means that two significant dependencies were shifted to the concrete job depending on these two abstract mechanisms rather than them depending on things about the job (parent batch management) or depending on each other (`Orchestration` calling `OrchestrationExecutor.execute`) so IMO moving these as such was a big win for keeping the dependency map between concepts optimally structured... Personally I find it a lot easier to mentally navigate around the files now but that would probably be the case regardless since I just worked on them a lot, so hard to say experientially but I think Metz would be pleased :rofl: